### PR TITLE
Add support for include_roles in guild prune

### PR DESCRIPTION
**Pull request by Khaaz**
Ref: discord/discord-api-docs#1563 

Add support for the includeRoles parameter in `getPruneCount` and `pruneMembers`.
I also fixed typings and jsdoc as all fields are optional.

⚠️  This is a breaking change as `includeRoles` is added before `reason` in pruneMembers.

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -869,26 +869,30 @@ class Client extends EventEmitter {
     /**
     * Get the prune count for a guild
     * @arg {String} guildID The ID of the guild
-    * @arg {Number} days The number of days of inactivity to prune for
+    * @arg {Number} [days] The number of days of inactivity to prune for. Default to 7
```
Comment (abalabahaha) [inline lib/Client.js]: `[days=7]` to show default values, not another sentence

**Review by abalabahaha — CHANGES_REQUESTED** [review]
Would be a good time to merge `days`/`reason`/`includeRoles` into an `options` object

File: lib/Client.js (hunk 1 above)
Comment (Khaaz) [reply lib/Client.js]: I thought we would only use those if it defaulted in the code. But I assume it's the same effect anyway.
```suggestion
    * @arg {Number} [days=7] The number of days of inactivity to prune for
```

**Comment by Khaaz** [comment]
Yeah I hesitated doing that as well. Do we want to do it for both getPruneCount and pruneMembers?
I'd assume we want but that makes it a breaking change for both method instead of only one

**Comment by abalabahaha** [comment]
Yes

File: index.d.ts [hunk 2]
```diff
@@ -1065,8 +1065,8 @@ declare namespace Eris {
     editRole(guildID: string, roleID: string, options: RoleOptions, reason?: string): Promise<Role>; // TODO not all options are available?
     editRolePosition(guildID: string, roleID: string, position: number): Promise<void>;
     deleteRole(guildID: string, roleID: string, reason?: string): Promise<void>;
-    getPruneCount(guildID: string, days: number): Promise<number>;
-    pruneMembers(guildID: string, days: number, reason?: string): Promise<number>;
+    getPruneCount(guildID: string, options: GetPruneOptions): Promise<number>;
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    getPruneCount(guildID: string, options?: GetPruneOptions): Promise<number>;
```

File: index.d.ts [hunk 3]
```diff
@@ -1065,8 +1065,8 @@ declare namespace Eris {
...
+    pruneMembers(guildID: string, options: PruneMemberOptions): Promise<number>;
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    pruneMembers(guildID: string, options?: PruneMemberOptions): Promise<number>;
```

File: index.d.ts [hunk 4]
```diff
@@ -1396,8 +1396,8 @@ declare namespace Eris {
     editEmoji(emojiID: string, options: { name: string; roles?: string[] }, reason?: string): Promise<Emoji>;
     deleteEmoji(emojiID: string, reason?: string): Promise<void>;
     createRole(options: RoleOptions | Role, reason?: string): Promise<Role>;
-    getPruneCount(days: number): Promise<number>;
-    pruneMembers(days: number, reason?: string): Promise<number>;
+    getPruneCount(options: GetPruneOptions): Promise<number>;
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    getPruneCount(options?: GetPruneOptions): Promise<number>;
```

File: index.d.ts [hunk 5]
```diff
@@ -1396,8 +1396,8 @@ declare namespace Eris {
...
+    pruneMembers(options: PruneMemberOptions): Promise<number>;
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    pruneMembers(options?: PruneMemberOptions): Promise<number>;
```

File: index.d.ts [hunk 6]
```diff
@@ -1703,6 +1703,15 @@ declare namespace Eris {
     };
   }
 
+  interface GetPruneOptions {
+    days?: number,
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    days?: number;
```

File: index.d.ts [hunk 7]
```diff
@@ -1703,6 +1703,15 @@ declare namespace Eris {
...
+    includeRoles?: string[]
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    includeRoles?: string[];
```

File: index.d.ts [hunk 8]
```diff
@@ -1703,6 +1703,15 @@ declare namespace Eris {
...
+  }
+
+  interface PruneMemberOptions extends GetPruneOptions {
+    reason?: string
```
Comment (Skillz4Killz) [inline index.d.ts]:
```suggestion
    reason?: string;
```

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**